Adds nova_compute_nvme_cleanup service in nova_compute container

auniyal61 opened the pull request.

auniyal61 commented on roles/edpm_nova/templates/nova_compute_nvme_cleanup.service.j2:
```diff
@@ -0,0 +1,13 @@
+[Service]
+Type=simple
+Restart=always
+ExecStart=/usr/bin/podman exec nova_compute /usr/share/openstack-nova/contrib/clean-on-delete --nvme
```
I think if it exists for any/unknown reason `Restart=always` will restart it - so this is enough.

also found we can add `RestartSec=10s`, to avoid crashloop on restart - wait for sometime before restart.

SeanMooney replied:
i don't think we want to do podman exec, doing that often used to cause issues with contaiernd which is why the old way of doign health checsk was so problematic.

what we shoudl do instead is deploy a second copy of the nova-compute  container image but with a different entry point that uses the clean-on-delete  script

SeanMooney replied:
by the way we shoudl do this via a seperate kolla config.json file for this so that the actul pid 1 in the contaienr is dumb-init which will reap any exitied process properly and the only thing ath dumb-init will end up running after the normal kolla_start stuff that copies the config and set permission will be the clean-on-delete script.

auniyal61 commented:
right now getting below error in CI

```

TASK [Assert systemd unit is running nova_compute_nvme_cleanup.service] ********
Monday 25 August 2025  10:39:23 +0000 (0:00:00.224)       0:00:06.846 *********
[WARNING]: conditional statements should not include jinja2 templating
delimiters such as {{ }} or {% %}. Found: unit_running.stdout in "{{
item.active | default(['active']) | list }}"
fatal: [compute-1]: FAILED! => changed=false
  assertion: unit_running.stdout in "{{ item.active | default(['active']) | list }}"
  evaluated_to: false
  msg: Systemd unit nova_compute_nvme_cleanup.service is not running
  
```

I think, service is not started because script is not present in correct path in container yet  `/usr/share/openstack-nova/contrib/clean-on-delete`

this should be corrected and passed once this is merged https://review.rdoproject.org/r/q/topic:nova-contrib-packaging

gibizer commented:
please link it to https://issues.redhat.com/browse/OSPRH-16552

auniyal61 commented:
I do not want to lose this patch entirely. 
created new (inprogress) [patch](https://github.com/openstack-k8s-operators/edpm-ansible/pull/1016) with addressing review comments.
and abandoning this as patchset in GitHub UI doesn’t show a clear diff.
